skrub/_gap_encoder.py

```diff
@@ -379,7 +378,13 @@ def get_feature_names_out(
         """
 
         vectorizer = CountVectorizer()
-        vectorizer.fit(list(self.H_dict_.keys()))
+        try:
+            vectorizer.fit(list(self.H_dict_.keys()))
+        except ValueError:
+            # The vectorizer failed to find words, we need to switch to
+            # char-level representation
+            vectorizer = CountVectorizer(analyzer="char_wb")
```

Comment:
I wonder if we should have a warning here. For users using the GapEncoder directly, an empty vocabulary probably indicates a problem eg they misunderstood what the analyzer they chose does, they applied it to the wrong column etc. OTOH users that use the GapEncoder inside the TableVectorizer probably don't care that it had to make adjustments for a specific column

Reply:
alright, LGTM then!